ocaml/xapi/xapi_observer.ml

```diff
@@ -194,20 +194,21 @@ let assert_valid_attributes attributes =
     )
     attributes
 
-let register_components ~__context ~self ~host =
+let default_attributes ~__context ~self ~host ~name_label =
   let pool = Helpers.get_pool ~__context in
   let host_label = Db.Host.get_name_label ~__context ~self:host in
   let host_uuid = Db.Host.get_uuid ~__context ~self:host in
   let pool_uuid = Db.Pool.get_uuid ~__context ~self:pool in
+  ("xs.pool.uuid", pool_uuid)
```

Comment:
This could be written more conventionally as a list of key-value pairs.

Reply:
Not sure what you mean, this is already a list of key-value pairs (or association list), do you mean using `[(k1, v1); ...]`?

Reply:
This is written as `a :: b :: [c]` but why not as `[a; b; c]`?

Reply:
It's preserving the original code, I guess the other style of writing lists was more in vogue back then

Reply:
Yeah I just blindly copied the old code, but I agree [a; b; c] is nicer :)

Reply:
>I guess the other style of writing lists was more in vogue back then

You mean, six months ago? ;)

Reply:
You have to write it using `::` if you prepend to an existing list; but here we know the entire list so I see no good reason to use `a :: b :: [c]`. Anyway, this is a minor point since it will be identical after parsing.